File: doggo-marketplace/marketplace-api/src/main/java/com/pets/service/PetService.java
package com.pets.service;

import com.pets.dto.PetRequest;
import com.pets.model.Pet;
import com.pets.repository.PetRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PetService {
    
    @Autowired
    private PetRepository petRepository;
    
    @Autowired
    private UserService userService;

    public List<Pet> getAllPets() {
        return petRepository.findAll();
    }

    public Pet getPetById(Integer id) {
        return petRepository.findByPId(id)
                .orElseThrow(() -> new RuntimeException("Pet not found"));
    }

    public Pet createPet(PetRequest request, Integer sellerId) {
        Pet pet = new Pet();
        pet.setPName(request.getName());
        pet.setPAge(request.getAgeMonths());
        pet.setGender(request.getGender().toString());
        pet.setPPhoto(request.getImgUrl());
        pet.setIsVaccinated(request.getVaccinated());
        pet.setFid(request.getFatherBreedId());
        pet.setMid(request.getMotherBreedId());
        pet.setSellerId(sellerId);

        return petRepository.save(pet);
    }

    public Pet updatePet(Integer id, PetRequest request, Integer sellerId) {
        Pet pet = getPetById(id);
        
        // Check if the user is the seller
        if (!pet.getSellerId().equals(sellerId)) {
            throw new RuntimeException("Not authorized to update this pet");
        }

        pet.setPName(request.getName());
        pet.setPAge(request.getAgeMonths());
        pet.setGender(request.getGender().toString());
        pet.setPPhoto(request.getImgUrl());
        pet.setIsVaccinated(request.getVaccinated());
        pet.setFid(request.getFatherBreedId());
        pet.setMid(request.getMotherBreedId());

        return petRepository.save(pet);
    }

    public void deletePet(Integer id, Integer sellerId) {
        Pet pet = getPetById(id);
        
        // Check if the user is the seller
        if (!pet.getSellerId().equals(sellerId)) {
            throw new RuntimeException("Not authorized to delete this pet");
        }

        petRepository.delete(pet);
    }

    public List<Pet> searchPets(String searchTerm) {
        return petRepository.searchPetsByName(searchTerm);
    }

    public List<Pet> getPetsByBreed(Integer breedId) {
        return petRepository.findByFid(breedId);
    }

    public List<Pet> getPetsBySeller(Integer sellerId) {
        return petRepository.findBySellerId(sellerId);
    }
}

File: doggo-marketplace/frontend/src/services/petService.js
import { marketplaceApi } from './api'

export const petService = {
  // Get all available pets
  async getAllPets() {
    try {
      const response = await marketplaceApi.get('/api/pets')
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to fetch pets')
    }
  },

  // Get pet by ID
  async getPetById(id) {
    try {
      const response = await marketplaceApi.get(`/api/pets/${id}`)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to fetch pet')
    }
  },

  // Create new pet listing
  async createPet(petData, sellerId) {
    try {
      const response = await marketplaceApi.post(`/api/pets?sellerId=${sellerId}`, petData)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to create pet listing')
    }
  },

  // Update pet listing
  async updatePet(id, petData, sellerId) {
    try {
      const response = await marketplaceApi.put(`/api/pets/${id}?sellerId=${sellerId}`, petData)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to update pet listing')
    }
  },

  // Delete pet listing
  async deletePet(id, sellerId) {
    try {
      await marketplaceApi.delete(`/api/pets/${id}?sellerId=${sellerId}`)
      return true
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to delete pet listing')
    }
  },

  // Search pets
  async searchPets(searchTerm) {
    try {
      const response = await marketplaceApi.get(`/api/pets/search?q=${encodeURIComponent(searchTerm)}`)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to search pets')
    }
  },

  // Get pets by breed
  async getPetsByBreed(breed) {
    try {
      const response = await marketplaceApi.get(`/api/pets/breed/${encodeURIComponent(breed)}`)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to fetch pets by breed')
    }
  },

  // Get pets by seller
  async getPetsBySeller(sellerId) {
    try {
      const response = await marketplaceApi.get(`/api/pets/seller/${sellerId}`)
      return response.data
    } catch (error) {
      throw new Error(error.response?.data?.error || 'Failed to fetch pets by seller')
    }
  }
} 